multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ApplicationWaitAfterStopHandler.java

```diff
@@ -0,0 +1,52 @@
+package org.cloudfoundry.multiapps.controller.process.util;
+
+import java.time.Duration;
+import java.util.Objects;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.steps.ProcessContext;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+
+@Named
+public class ApplicationWaitAfterStopHandler {
+
+    protected ApplicationWaitAfterStopVariableGetter delayVariableGetter;
+
+    @Inject
+    public ApplicationWaitAfterStopHandler(ApplicationWaitAfterStopVariableGetter delayVariableGetter) {
+        this.delayVariableGetter = delayVariableGetter;
+    }
+
+    public boolean shouldWaitAfterAppStop(ProcessContext context) {
+        if (!delayVariableGetter.isAppStopDelayEnvVariableSet(context)) {
+            return false;
+        }
+        Duration durationToWait = delayVariableGetter.getDelayDurationFromAppEnv(context);
+        return !durationToWait.isZero();
+    }
+
+    public void setDelayAfterAppStop(ProcessContext context, String appName) {
```

Comment:
This method is quite strange because it is named with `set` prefix and I am expecting to see to pass delay value to the method. Either change the name of the method or pass delay value from outside the method as parameter.